docs/QuestionsIssuesPrs.md

```diff
@@ -0,0 +1,109 @@
+   you can always post a message to check if anybody is still working on that issue.
+
+5. Create a branch with the name that follows the [naming conventions for branch names](FormatsAndConventions.md#branch).
+
+6. Fix the issue in the new branch. 
+   * Remember to adhere to coding style and testing requirements of the project, as specified in the project 
+     documentation.
+   * A PR should bring the code from one complete state to another. That means when you update code, relevant
+     code comments, tests, user docs, and developer docs should be updated too.
+   * Commit code at suitable points. Follow the [conventions for commit messages](FormatsAndConventions.md#commit).
+
+7. When you think you are done with fix, create a new PR. Here are the steps: 
+   1. **Update a PR with upstream changes**: There is a chance that While you were write code for the PR, 
+      the master branch of the _upstream repo_ (i.e. the original repo you forked from) has been updated with new code. 
+      In that case you need to update your fork and your clone with that new code so that the only difference between 
+      the code in your branch and the upstream master branch is the code you wrote.  <br>
+      
+      > See: [How to update a PR with upstream changes](HowToGuides.md#update-a-pr-with-upstream-changes)
+   
+   2. **Get _local green_**: 
+      
+      * Ensure that the tests are passing in your local environment. 
+      * If the project uses any other static analysis tools (e.g. checkstyle), use them to detect any potential problems
+        in the new code.
+   1. **Push the code of your branch to your fork**.
+   2. **Create a PR**: 
+      
+      * Follow the [naming conventions for PRs](FormatsAndConventions.md#pr).
+      * Include `Fixes #IssueNumber` (e.g. `Fixes #1234`) in the PR description so that GitHub can auto-link the 
+      relevant issue and 
+      [auto-close the issue when the PR is merged](https://help.github.com/articles/closing-issues-via-commit-messages/).
+      You can look at [this PR](https://github.com/TEAMMATES/teammates/pull/6200) for an example.
+      * Before submitting the PR, go through the diff (i.e. the modifications included in your PR) shown by GitHub
+        to ensure that it does not contain any changes that are not relevant to the PR. If there are any, 
+        you need to update your branch to remove those changes.
+   
+     > You may create a PR even before you are done with the fix, if you want to seek some early feedback from 
+     > the dev team.
+
+8. After creating the PR, 
+   1. Wait for CI (i.e. Travis, AppVeyor) to run tests/checks against your PR and report status. 
+      If any errors are reported, fix those problems and push the fixes to the same branch.
+   2. When there are no more CI errors, post a comment to inform the dev team that the PR is ready for review. 
```

Comment:
@damithc 
> It does some effort to find the devs to mention, yes?

Yes it does require some effort, either by manually browsing the history or installing some git utility scripts. But if you think about it, we (those handling the PRs) are already doing it anyway. e.g. when you pick reviewers like in https://github.com/se-edu/addressbook-level4/pull/202#issuecomment-266179194

This change:
> Also @mention developers (even those not part on the core dev team) who have touched the same area of the code as the PR.

would push this effort to the *contributors*, thus making it easier for the core devs :-)

@damithc 
> Sounds like a job for this bot :-p

Yes, that would help.